pkg/rehearse/metrics.go

```diff
@@ -0,0 +1,131 @@
+package rehearse
+
+import (
+	"encoding/json"
+	"github.com/openshift/ci-operator-prowgen/pkg/config"
+	"github.com/sirupsen/logrus"
+	"io/ioutil"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/test-infra/prow/apis/prowjobs/v1"
+	prowconfig "k8s.io/test-infra/prow/config"
+)
+
+type ExecutionMetrics struct {
+	SubmittedRehearsals sets.String
+	FailedRehearsals    sets.String
+	PassedRehearsals    sets.String
+}
+
+func (m *ExecutionMetrics) RecordFailedRehearsal(job *v1.ProwJob) {
```

Comment:
If the fields are exported (as they need to be to serialize) we don't really need the helpers

Reply:
I like the readability at the callsite though - I don't like our `main()` cluttered with modifying the internal state of some random struct.

Reply:
How is it less readable to add to a map as a field?